src/components/ens-profile/ActionButtons/MoreButton.tsx

```diff
@@ -0,0 +1,140 @@
+import { formatAddressForDisplay } from '@rainbow-me/utils/abbreviations';
+
+const ACTIONS = {
+  ADD_CONTACT: 'add-contact',
+  COPY_ADDRESS: 'copy-address',
+  ETHERSCAN: 'etherscan',
+  REMOVE_CONTACT: 'remove-contact',
+};
+
+export default function MoreButton({
+  address,
+  ensName,
+}: {
+  address?: string;
+  ensName?: string;
+}) {
+  const { navigate } = useNavigation();
+  const { setClipboard } = useClipboard();
+  const { contacts, onRemoveContact } = useContacts();
+
+  const contact = useMemo(
+    () => (address ? contacts[address.toLowerCase()] : undefined),
+    [address, contacts]
+  );
+
+  const formattedAddress = useMemo(
+    () => (address ? formatAddressForDisplay(address, 4, 4) : ''),
+    [address]
+  );
+
+  const menuItems = useMemo(() => {
+    return [
+      {
+        actionKey: ACTIONS.COPY_ADDRESS,
+        actionTitle: lang.t('profiles.details.copy_address'),
+        discoverabilityTitle: formattedAddress,
+        icon: {
+          iconType: 'SYSTEM',
+          iconValue: 'square.on.square',
+        },
+      },
+      contact
+        ? {
+            actionKey: ACTIONS.REMOVE_CONTACT,
+            actionTitle: lang.t('profiles.details.remove_from_contacts'),
+            icon: {
+              iconType: 'SYSTEM',
+              iconValue: 'person.crop.circle.badge.minus',
+            },
+          }
+        : {
+            actionKey: ACTIONS.ADD_CONTACT,
+            actionTitle: lang.t('profiles.details.add_to_contacts'),
+            icon: {
+              iconType: 'SYSTEM',
+              iconValue: 'person.crop.circle.badge.plus',
+            },
+          },
+      {
+        actionKey: ACTIONS.ETHERSCAN,
+        actionTitle: lang.t('profiles.details.view_on_etherscan'),
+        icon: {
+          iconType: 'SYSTEM',
+          iconValue: 'link',
+        },
+      },
+    ] as MenuActionConfig[];
+  }, [contact, formattedAddress]);
+
+  const handlePressMenuItem = useCallback(
+    ({ nativeEvent: { actionKey } }) => {
+      if (actionKey === ACTIONS.COPY_ADDRESS) {
+        setClipboard(address);
+      }
+      if (address && actionKey === ACTIONS.ETHERSCAN) {
+        ethereumUtils.openAddressInBlockExplorer(address);
+      }
+      if (actionKey === ACTIONS.ADD_CONTACT) {
+        navigate(Routes.MODAL_SCREEN, {
+          address,
+          contact,
+          ens: ensName,
+          nickname: ensName,
+          type: 'contact_profile',
+        });
+      }
+      if (actionKey === ACTIONS.REMOVE_CONTACT) {
+        showDeleteContactActionSheet({
+          address,
+          nickname: contact.nickname,
+          removeContact: onRemoveContact,
+        });
+        android && Keyboard.dismiss();
+      }
+    },
+    [address, contact, ensName, navigate, onRemoveContact, setClipboard]
+  );
+
+  const handleAndroidPress = useCallback(() => {
+    const actionSheetOptions = menuItems
+      .map(item => item?.actionTitle)
+      .filter(x => x) as any;
```

Comment:
```suggestion
      .filter(Boolean) as any;
```

Reply:
also added this change everywhere we were doing it 👌